client/src/components/MultiJurisdictionAudit/RoundManagement/generateSheets.ts

```diff
@@ -12,6 +12,44 @@ const LABEL_MARGIN_X = 9
 const LABEL_PADDING_Y = 7
 const LABEL_PADDING_X = 7
 
+const generateLabelPages = (labels: jsPDF, ballots: IBallot[]): string[][][] =>
+  ballots.reduce(
+    (a, ballot, ballotIndex) => {
+      const lines: string[] = [
+        `${ballot.auditBoard!.name} - Ballot Number: ${ballot.position}`,
+        ballot.batch.container && `Container: ${ballot.batch.container}`,
+        ballot.batch.tabulator && `Tabulator: ${ballot.batch.tabulator}`,
+        `Batch: ${ballot.batch.name}`,
+        ballot.imprintedId !== undefined
+          ? `Imprinted ID: ${ballot.imprintedId}`
+          : null,
+      ].filter(line => line) as string[] // ts is not seeing us filtering out the nulls
+      const finalLines: string[] = []
+      lines.forEach(line => {
+        if (lines.length < 8) {
+          finalLines.push(
+            ...labels.splitTextToSize(line, LABEL_WIDTH - LABEL_PADDING_X * 2)
+          )
+        } else {
+          finalLines.push(
+            ...labels.splitTextToSize(
+              line,
+              LABEL_WIDTH - LABEL_PADDING_X * 2
+            )[0]
+          )
+        }
+      })
```

Comment:
since you're using functional style in the outer loop with `reduce`, would be cleaner to stick to that and use a `map` on `lines`.